devito/types/dense.py

```diff
@@ -1201,21 +1217,20 @@ class TimeFunction(Function):
     _time_position = 0
     """Position of time index among the function indices."""
 
-    def __init__(self, *args, **kwargs):
-        if not self._cached():
-            self.time_dim = kwargs.get('time_dim', self.dimensions[self._time_position])
-            self._time_order = kwargs.get('time_order', 1)
-            super(TimeFunction, self).__init__(*args, **kwargs)
+    def __init_finalize__(self, *args, **kwargs):
```

Comment:
note for reviewers: this is exactly like before, but de-indented as we don't need ``if not self._cached()`` anymore